domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -2224,51 +2221,206 @@ public void getDistinctOn(TestContext context) throws IOException {
-      add(new FacetField("jsonb->>'edition'"));
-    }};
+    List<FacetField> facets = new ArrayList<FacetField>() {
+      {
+        add(new FacetField("jsonb->>'edition'"));
+      }
+    };
     String distinctOn = "jsonb->>'order_format'";
     //with facets and return count
-    Async async1 = context.async();
+    Async async = context.async();
     postgresClient.get(MOCK_POLINES_TABLE, Object.class, "*", "", true, false,
       false, facets, distinctOn, handler -> {
         ResultInfo resultInfo = handler.result().getResultInfo();
         context.assertEquals(4, resultInfo.getTotalRecords());
         List<Facet> retFacets = resultInfo.getFacets();
         context.assertEquals(1, retFacets.size());
-        async1.complete();
+        async.complete();
       });
-    async1.awaitSuccess();
+    async.awaitSuccess();
 
-    String whereClause =  "WHERE jsonb->>'order_format' = 'Other'";
+    String whereClause = "WHERE jsonb->>'order_format' = 'Other'";
 
     //with facets and where clause RMB-355
-    Async async2 = context.async();
     postgresClient.get(MOCK_POLINES_TABLE, Object.class, "*", whereClause, true, false,
       false, facets, distinctOn, handler -> {
         ResultInfo resultInfo = handler.result().getResultInfo();
         context.assertEquals(1, resultInfo.getTotalRecords());
         List<Facet> retFacets = resultInfo.getFacets();
         context.assertEquals(1, retFacets.size());
-        async2.complete();
+        async.complete();
+      });
+    async.awaitSuccess();
+  }
+
+  @Test
+  public void getCQLWrapperFailure(TestContext context) throws IOException, FieldException {
+    final String tableDefiniton = "id UUID PRIMARY KEY , jsonb JSONB NOT NULL, distinct_test_field TEXT";
+    createTableWithPoLines(context, MOCK_POLINES_TABLE, tableDefiniton);
+    CQL2PgJSON cql2pgJson = new CQL2PgJSON(MOCK_POLINES_TABLE + ".jsonb");
+    CQLWrapper cqlWrapper = new CQLWrapper(cql2pgJson,
+      "cql.allRecords="); // syntax error
+    postgresClient.get(MOCK_POLINES_TABLE, Object.class, "*",
+      cqlWrapper, true, true, false, null, null, context.asyncAssertFailure());
+  }
+
+  @Test
+  public void getCQLWrapperNoCount(TestContext context) throws IOException, FieldException {
+    final String tableDefiniton = "id UUID PRIMARY KEY , jsonb JSONB NOT NULL, distinct_test_field TEXT";
+    createTableWithPoLines(context, MOCK_POLINES_TABLE, tableDefiniton);
+
+    CQL2PgJSON cql2pgJson = new CQL2PgJSON(MOCK_POLINES_TABLE + ".jsonb");
+
+    CQLWrapper cqlWrapper = new CQLWrapper(cql2pgJson, "cql.allRecords=1");
+    postgresClient.get(MOCK_POLINES_TABLE, Object.class, "*",
+      cqlWrapper, false, true, false, null, null/*facets*/, handler -> {
+        context.assertTrue(handler.succeeded());
+        ResultInfo resultInfo = handler.result().getResultInfo();
+        context.assertEquals(6, resultInfo.getTotalRecords());
+        context.asyncAssertSuccess();
```

Comment:
This handler will never be executed, the unit test will succeed without running the asserts.

See https://vertx.io/docs/vertx-unit/java/#_asynchronous_assertions how to do it:

    postgresClient.get(MOCK_POLINES_TABLE, Object.class, "*",
      cqlWrapper, false, true, false, null, null/*facets*/, context.asyncAssertSuccess(result -> {
        ResultInfo resultInfo = handler.result().getResultInfo();
        context.assertEquals(6, resultInfo.getTotalRecords());
      }});